### PR TITLE
dmfarcas: hamburger mobile menu and banner fix

### LINKED ISSUE #96: Enhance header display on mobile devices - hamburger menu
Current header behavior on mobile:
Screenshot_20200323-174215_Chrome

Should be changed to:
Screenshot_20200323-174224_Chrome

### PR DESCRIPTION
**Pull request by dmfarcas**
Closes #96 

url that leads to feature: https://taskforce-fe-components-git-fork-dmfarcas-dmfarcas.code4ro.now.sh/iframe.html?id=header--with-name

### CODE REVIEW

File: src/components/banner/banner.scss [hunk 1]
```diff
@@ -24,11 +24,17 @@ $info-card-colors: (
   display: flex;
   justify-content: center;
 
+  padding-left: 1.25rem;
```
Comment (dmfarcas) [inline src/components/banner/banner.scss]: stole these from stirioficiale.ro

**Comment by dmfarcas** [comment]
https://taskforce-fe-components-git-fork-dmfarcas-dmfarcas.code4ro.now.sh/iframe.html?id=header--with-name

File: src/components/banner/banner.scss [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
 
 $info-card-colors: (
-  "red": #ff1616,
-  "pink": #f5bcd6,
-  "green": #05d6c7,
-  "yellow": #ffde59
+  'red': #ff1616,
+  'pink': #f5bcd6,
+  'green': #05d6c7,
+  'yellow': #ffde59
```
[the remark this answers is not in the document]
Comment (dmfarcas) [reply src/components/banner/banner.scss]: fixed :D
btw, do you automatically style CSS with a tool? if so, which one?

[the remark this answers is not in the document]
Comment (moonflare) [reply src/components/banner/banner.scss]: You mean formatting of the files? We don't at the moment. We should probably do that to keep consistency with JavaScript files as well and avoid multiple code styles

**Comment by dmfarcas** [comment]
> Thanks for the PR 🙏 I left a few comments 💬

thank you too! done

File: src/components/header/header.js [hunk 3]
```diff
@@ -1,18 +1,31 @@
-import React from "react";
+import React, { useState } from "react";
 import PropTypes from "prop-types";
 import "./header.scss";
 
+import Hamburger from "../../images/icons/hamburger.svg";
+import HamburgerOpen from "../../images/icons/hamburger-open.svg";
+
 export const Header = ({ name, Logo, MenuItems, ProfileItems }) => {
+  const [hamburgerOpened, hamburgerOnClick] = useState(true);
```
Comment (Utwo) [inline src/components/header/header.js]: I think by default, the menu should be closed

Comment (moonflare) [inline src/components/header/header.js]:
```suggestion
  const [isOpen, setIsOpen] = useState(false);
```

File: src/components/header/header.js [hunk 4]
```diff
@@ -1,18 +1,31 @@
...
-        {name ? <span className="App-header__title">{name}</span> : ""}
+        {name && <span className="App-header__title">{name}</span>}
       </div>
-      <nav>
+      {
+        <div className="App-header__hamburger">
+          <HamburgerIcon onClick={() => hamburgerOnClick(!hamburgerOpened)} />
+        </div>
+      }
+      {/* {hamburgerOpened && ( */}
```
Comment (moonflare) [inline src/components/header/header.js]: Please remove the comments

File: src/components/header/header.js [hunk 5]
```diff
@@ -1,18 +1,31 @@
...
+
   return (
     <header className="App-header container">
       <div className="App-header__logo">
         {Logo}
-        {name ? <span className="App-header__title">{name}</span> : ""}
+        {name && <span className="App-header__title">{name}</span>}
       </div>
-      <nav>
+      {
```
Comment (moonflare) [inline src/components/header/header.js]: We don't need the `{}` for this block

File: src/components/header/header.js [hunk 6]
```diff
@@ -1,18 +1,31 @@
...
+        <div className="App-header__hamburger">
```
Comment (moonflare) [inline src/components/header/header.js]: nit
```suggestion
        <div className="App-header__menu-icon">
```

File: src/components/header/header.scss [hunk 7]
```diff
@@ -44,13 +50,34 @@
...
+      display: block;
+      text-align: right;
+      user-select: none;
+    }
+
+    .App-header__logo {
+      flex: 0 0 75%;
+    }
+
+    svg {
```
Comment (moonflare) [inline src/components/header/header.scss]: Can we please target the menu icon with a class? Just so we avoid side effects on the styles

Comment (moonflare) [reply src/components/header/header.scss]: nit: We could also remove hover effect and cursor pointer on the links. Should not be required on mobile

File: src/components/header/header.js (hunk 6 above)
Comment (moonflare) [reply src/components/header/header.js]: In general I would use terminology that describes the elements, not necessarily the type of the "element" in this case.

File: src/components/header/header.scss (hunk 7 above)
Comment (dmfarcas) [reply src/components/header/header.scss]: not exactly sure how to handle this. media query?

> nit: We could also remove hover effect and cursor pointer on the links. Should not be required on mobile

Comment (moonflare) [reply src/components/header/header.scss]: Yes, in the same breakpoint query, inside the nav element, afte [this line](https://github.com/code4romania/taskforce-fe-components/pull/102/files#diff-748c6711719a7c21d1a4684ae78c3987R74) just have something below.
```css
nav {
    a:hover { 
        cursor: auto;
        background: none;
    }
}
```

**Pull request merged**